tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1094,6 +1098,70 @@ const reducer = createReducer(
       singleSelectionHeaders: newOrder,
     };
   }),
+  on(actions.dataTableColumnEdited, (state, {fobState, newHeaders}) => {
+    const enabledNewHeaders: ColumnHeader[] = [];
+    const disabledNewHeaders: ColumnHeader[] = [];
+
+    newHeaders.forEach((header) => {
+      if (header.enabled) {
+        enabledNewHeaders.push(header);
+      } else {
+        disabledNewHeaders.push(header);
+      }
+    });
+
+    if (fobState === FobState.RANGE) {
+      return {
+        ...state,
+        rangeSelectionHeaders: enabledNewHeaders.concat(disabledNewHeaders),
+      };
+    }
+
+    return {
+      ...state,
+      singleSelectionHeaders: enabledNewHeaders.concat(disabledNewHeaders),
+    };
+  }),
+  on(actions.dataTableColumnToggled, (state, {fobState, headerType}) => {
+    const targetedHeaders =
+      fobState === FobState.RANGE
+        ? state.rangeSelectionHeaders
+        : state.singleSelectionHeaders;
+
+    const toggledHeaderCurrentIndex = targetedHeaders.findIndex(
+      (element) => element.type === headerType
+    );
+
+    // If the header is being enabled it goes at the bottom of the currently
+    // enabled headers. If it is being disabled it goes to the top of the
+    // currently disabled headers.
+    let newToggledHeaderIndex = getEnabledCount(targetedHeaders);
+    if (targetedHeaders[toggledHeaderCurrentIndex].enabled) {
+      newToggledHeaderIndex--;
+    }
+    const newHeaders = moveHeader(
+      toggledHeaderCurrentIndex,
+      newToggledHeaderIndex,
+      targetedHeaders
+    );
+
+    newHeaders[newToggledHeaderIndex] = {
```

Comment:
I found it a little amusing that you would spend all that time moving the toggled header and then completely overwrite it with a new value. I was going to suggest:

`newHeaders[newToggledHeaderIndex].enabled = !newHeaders[newToggledHeaderIndex].enabled`

But are you doing it this way so that you don't modify the old state?

Reply:
That is correct. This is to keep it a "pure" function which makes no changes to the current state.